discovery-infra/test_infra/helper_classes/proxy_params.py

```diff
@@ -0,0 +1,29 @@
+from test_infra.helper_classes.idict import IDict
+
+
+class Proxy(IDict):
```

Comment:
Few things:

1. Keep class name and module name the same
2. If there is no complicated logic inside `Proxy` decorate it with dataclass
3. If it's a dataclass you don't need IDict and can simply `from dataclasses import dataclass, asdict`
4. Why do we need this class? `models.Proxy` gives us the same functionality

Reply:
1. The problem is that `models.Proxy` is not `pickleable`  so doing a deepcopy can cause issues
2. Changing
3. Wanted to keep the same logic as the existing class

Reply:
@nshidlin You can `proxy.to_dict()`. 
e.g.
```python
from assisted_service_client import models
proxy = models.Proxy(http_proxy="http_proxy", https_proxy="https_proxy", no_proxy="no_proxy")
print(proxy.to_dict())

---
{'http_proxy': 'http_proxy', 'https_proxy': 'https_proxy', 'no_proxy': 'no_proxy'}

```